featuretools/primitives/standard/transform/natural_language/num_words.py

```diff
@@ -25,6 +32,12 @@ class NumWords(TransformPrimitive):
 
     def get_function(self):
         def word_counter(array):
-            return array.fillna("").str.count(" ") + 1
+            def _get_number_of_words(elem: pd.Series):
+                """Returns the number of words or null given non-iterable input"""
+                if not isinstance(elem, Iterable):
```

Comment:
One question we need to decide is what to return give null input. Prior to this change, we returned 0 given null input. If we choose to return null, we should update `NumCharacters` as well to keep the primitives uniform.